walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,154 @@
+    // Stores all TimeRanges in which optional attendees are busy with no overlaps.
+    ArrayList<TimeRange> notOptionalTimes = new ArrayList<>();
+    // Stores all TimeRanges in which mandatory attendees can meet.
+    ArrayList<TimeRange> availableTimes = new ArrayList<>();
+    // Stores all TimeRanges in which optional attendees can meet.
+    ArrayList<TimeRange> optionalCalendar = new ArrayList<>();
+
+    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+      return Arrays.asList();
+    }
+    // No events to conflict with.
+    if (events.size() == 0) {
+      return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+      
+    ArrayList<String> toDeleteMan = new ArrayList<>();
+    ArrayList<String> toDeleteOpt = new ArrayList<>();
+      
+    for (Event event : events) {
+      for (String attendee : event.getAttendees()) {
+        if (attendeesWithoutEvents.contains(attendee)) {
+          // Adds TimeRange if any mandatory attendee attends this event.
+          timesRequiredAttendees.add(event.getWhen());
+          // Adds whenever a mandatory attendee appears in an event. 
+          toDeleteMan.add(attendee);
+        }
+        if (optionalAttendeesWithoutEvents.contains(attendee)) {
+          timesOptionalAttendees.add(event.getWhen());
+          toDeleteOpt.add(attendee);
+        }
+      }
+    }
+    // Deletes from our initial attendee counts those who appears on events so we can check 
+    // whether there is no overlap between the attendees of our given Events and those request in Meeting Request.
+    attendeesWithoutEvents.removeAll(toDeleteMan);
+    optionalAttendeesWithoutEvents.removeAll(toDeleteOpt);
+    
+    // Transforming collections of TimeRanges into an ArrayList to allow indexing.
+    ArrayList<TimeRange> allTimesRequiredAttendees = new ArrayList<>(timesRequiredAttendees);
+    ArrayList<TimeRange> allTimesOptionalAttendees = new ArrayList<>(timesOptionalAttendees);
+    Collections.sort(allTimesRequiredAttendees, TimeRange.ORDER_BY_START);
+    Collections.sort(allTimesOptionalAttendees, TimeRange.ORDER_BY_START);
+
+    // No events conflict the schedules of the requested attendees. 
+    if (attendeesWithoutEvents.size() == initialAttendeeCount && optionalAttendeesWithoutEvents.size() == initialOptionalAttendeeCount) {
+      return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+    // Checks whether mandatory attendees have no restrictions in their schedules.
+    if (allTimesRequiredAttendees.isEmpty()) {
+      ArrayList<TimeRange> optionalAttendeesCalendar = getCalendar(
+                                                                    allTimesOptionalAttendees, 
+                                                                    notOptionalTimes, 
+                                                                    optionalCalendar, 
+                                                                  );
+      Collections.sort(optionalAttendeesCalendar, TimeRange.ORDER_BY_START);
```

Comment:
You always end up sorting the output of "getCalendar". Why not just sort the response before returning it from the function?